06_modules_lab/03.py

```diff
@@ -1,11 +1,30 @@
-""" Write a program that searches current working directory
-for files larger than 1MB. Every time you find such a file print
-its name to the user.
-
-- When the program finds a large file. It should ask the user
-  a message asking if she wants to delete it, and delete the
-  file if requested
-
-- Take threshold and path as command line arguments
-"""
-
+##################Part 3
+import os
+import sys
+import argparse
+
+parser = argparse.ArgumentParser(description='The program gets a path where file sizes over a pre-defined value are deleted, if required by the user')
+parser.add_argument('-p','--path', help='full path (\ - terminated) to search', required=True)
+parser.add_argument('-ms','--minsize', help='The size threshold', required=True)
+args = vars(parser.parse_args())
+#print(args)
+'''
+if len(sys.argv) != 3:
+    print('Usage: %s <path> ,minsize>' % sys.argv[0])
+    sys.exit(1)
+
+(_, path, minsize) = sys.argv
+'''
+#print('Starting at %s, looking for file sizes bigger than %d' % (args['path'], int(args['minsize'])) )
+
+for name in os.listdir(args['path']):
+    #print('%s'% name)
+    #statinfo = os.stat(path +  name)
+    #print(statinfo)
+    size = os.path.getsize(args['path'] + name)
+    if size > int(args['minsize']):
+        #print(size)
+        print('Delete %s? [1 for YES 0 for NO]' % (args['path'] + name))
+        ans = input()
+        if int(ans) == 1:#delete the file
+           os.remove(args['path'] + name)
```

Comment:
This is because you enter a ‘path’ not ending with ‘\’

From: Gabor Szabo [mailto:notifications@github.com]
Sent: Wednesday, July 06, 2016 9:57 AM
To: ynonp/python-examples-verint-2016-07
Cc: Rosenthal, Hanan (Embedded); Author
Subject: Re: [ynonp/python-examples-verint-2016-07] ToCode Solution: תרגול מודולים (#15)


In 06_modules_lab/03.py<https://github.com/ynonp/python-examples-verint-2016-07/pull/15#discussion_r69682575>:

> +

> +(_, path, minsize) = sys.argv

> +'''

> +#print('Starting at %s, looking for file sizes bigger than %d' % (args['path'], int(args['minsize'])) )

> +

> +for name in os.listdir(args['path']):

> +    #print('%s'% name)

> +    #statinfo = os.stat(path +  name)

> +    #print(statinfo)

> +    size = os.path.getsize(args['path'] + name)

> +    if size > int(args['minsize']):

> +        #print(size)

> +        print('Delete %s? [1 for YES 0 for NO]' % (args['path'] + name))

> +        ans = input()

> +        if int(ans) == 1:#delete the file

> +           os.remove(args['path'] + name)

I tried the program passing -p . -ms 1 and got an exception No such file or directory: '..git'
BTW I'd not convert the ans to number, I'd just compare it to the string '1'

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/ynonp/python-examples-verint-2016-07/pull/15/files/4c6e32148f38cbd75d4ac68a96347ee582ee4ba3..8ee551475b91ff1fbf389c1cba1eb0b2186848c5#r69682575>, or mute the thread<https://github.com/notifications/unsubscribe/ATVHXg5eYkGPK7dxjUlVROO4WjHpl2G1ks5qS1GqgaJpZM4JE3xJ>.

-- 
This electronic message may contain proprietary and confidential 
information of Verint Systems Inc., its affiliates and/or subsidiaries.
The information is intended to be for the use of the individual(s) or
entity(ies) named above. If you are not the intended recipient (or 
authorized to receive this e-mail for the intended recipient), you may not 
use, copy, disclose or distribute to anyone this message or any information 
contained in this message. If you have received this electronic message in 
error, please notify us by replying to this e-mail.